File: HackerRank/SepCodesprint2016/TwoCharacters.py
import sys
sys.stdin = open("/Users/seeva92/Workspace/Contests/1.txt", "r")
sys.stdout = open("/Users/seeva92/Workspace/Contests/2.txt", "w")
n = int(input())
string = input()
def compute():
	res = 0
	
		for i in range(0,26):
			for j in range(0,26):
				flag = True
				curr = 0
				pIdx = -1
				while(True):
					if flag:					
						pIdx = s.find(chr(ord('a')+i),pIdx+1)
						if pIdx == -1:
							break
						else:
							curr+=1
							flag = False
					else:
						pIdx = s.find(chr(ord('a')+j),pIdx+1)
						if pIdx == -1:
							break
						else:
							curr+=1
							flag = False					
				res = max(res,curr)			
				curr = 0
				flag = True			
				for k in s:
					if flag:
						if chr(ord('a')+j) != k:
							continue
						else:
							curr+=1
							flag = False
					else:
						if chr(ord('a')+i) != k:
							continue
						else:
							curr+=1
							flag = True

				res = max(res,curr)			
				# print(res)
	print(res)		


compute()

File: HackerRank/SepCodesprint2016/TwoCharacters.cpp
#include <bits/stdc++.h>
const int mod = 1e9 + 7;
using namespace std;
int n; string s;
int main() {
#ifndef ONLINE_JUDGE
	freopen("/Users/seeva92/Workspace/Contests/1.txt", "r", stdin);
	freopen("/Users/seeva92/Workspace/Contests/2.txt", "w", stdout);
#endif
	ios::sync_with_stdio(false);
	cin.tie(0);
	cin >> n;
	cin >> s;
	int res = 0;
	for (int i = 0; i < 26; i++) {
		for (int j = 0; j < 26; j++) {
			// cout << (char)('a' + i) << " " << (char)('a' + j) << '\n';
			if (i == j) continue;
			bool flag = true;
			int curr = 0;
			int vis = 0;
			for (int k = 0; k < n; k++ ) {
				if (s[k] == char('a' + i) || s[k] == char('a' + j)) {
					vis++;
					// cout << s[k];
					if (flag && s[k] == char('a' + i)) {
						curr++;
						flag = !flag;
					} else if (!flag && s[k] == char('a' + j)) {
						curr++;
						flag = !flag;
					} else break;
				}
			}
			// cout << char('a' + i) << " " << char('a' + j) << " " << vis << " " << curr << '\n';
			if (vis == curr)
				res = max(curr, res);
		}
	}
	if (res == 1) cout << 0 << '\n';
	else cout << res << '\n';
}
